controllers/etcdpeer_controller_test.go

```diff
@@ -1,62 +1,52 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
 package controllers
 
 import (
-	"context"
+	"github.com/avast/retry-go"
+	"testing"
 	"time"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/require"
 	appsv1 "k8s.io/api/apps/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	etcdv1alpha1 "github.com/improbable-eng/etcd-cluster-operator/api/v1alpha1"
 )
 
-var _ = Describe("Etcd peer controller", func() {
-	ctx := context.Background()
-	SetupTest(ctx)
-
-	Context("ReplicaSet creation", func() {
-		It("Should create a ReplicaSet if one does not exist already", func() {
-
-			By("Creating a EtcdPeer")
-			etcdPeer := etcdv1alpha1.EtcdPeer{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels:      make(map[string]string),
-					Annotations: make(map[string]string),
-					Name:        "bees",
-					Namespace:   "default",
-				},
-			}
-
-			err := k8sClient.Create(ctx, &etcdPeer)
-			Expect(err).NotTo(HaveOccurred(), "failed to create EtcdPeer resource")
-
-			By("The controller having created a ReplicaSet")
-			replicaSet := &appsv1.ReplicaSet{}
-			Eventually(
-				getResourceFunc(
-					ctx,
-					client.ObjectKey{
-						// Same name and namespace as the EtcdPeer above
-						Name:      etcdPeer.Name,
-						Namespace: etcdPeer.Namespace,
-					},
-					replicaSet,
-				),
-				// Timeout
-				time.Second*5,
-				// Polling Interval
-				time.Millisecond*500,
-			).Should(BeNil())
-		})
-	})
-})
+func (s *controllerSuite) testPeerController(t *testing.T) {
+	t.Run("TestPeerController_OnCreation_CreatesReplicaSet", func(t *testing.T) {
```

Comment:
Are the underscores special in test names here? Can't this just be normal text, i.e., "Peer controller creates replica set on creation".

Reply:
any name will do, its just what gets printed in the test output, i.e.:

```
=== RUN   TestAPIs
--- PASS: TestAPIs (8.86s)
=== RUN   TestAPIs/PeerControllers
    --- PASS: TestAPIs/PeerControllers (0.52s)
=== RUN   TestAPIs/PeerControllers/TestPeerController_OnCreation_CreatesReplicaSet
        --- PASS: TestAPIs/PeerControllers/TestPeerController_OnCreation_CreatesReplicaSet (0.52s)
PASS
```

i tend to prefer the `$testObject_$testAction_$expectedResult` format but it's flexible

Reply:
No real preference. I'm used to just prose, but something more formatted is also fine.